spec/hyacinth/adapters/external_identifier_adapter/datacite_spec.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite do
+  let(:datacite) { described_class.new }
+  let(:dod) do
+    data = JSON.parse(file_fixture('files/datacite/ezid_item.json').read)
+    data['identifiers'] = ['item.' + SecureRandom.uuid] # random identifer to avoid collisions
+    data
+  end
+  let(:metadata) do
+    { type: 'dois',
+      attributes: {
+        doi: '10.33555/0645-3z82',
+        state: 'draft',
+        titles: [{ title: 'The Good Title' }],
+        creators: [{ name: "Doe, Jane" }],
+        url: 'https://www.columbia.edu',
+        publisher: 'Self',
+        publicationYear: 2002,
+        types: { resourceTypeGeneral: 'Text' },
+        schemaVersion: 'http://datacite.org/schema/kernel-4',
+        prefix: '10.33555'
+      } }
+  end
+  let(:metadata_payload_json) do
+    '
+    {"data":
+       {"type":"dois",
+        "attributes":
+          {"prefix":"10.33555",
+           "creators":
+             [{"name":"Salinger, J. D."},{"name":"Lincoln, Abraham"}],
+           "titles":
+             [{"title":"The Catcher in the Rye"}],
+           "publisher":"The Best Publisher Ever",
+           "publicationYear":1951,
+           "types":
+             {"resourceTypeGeneral":"Image"},
+           "url":"https://www.columbia.edu",
+           "schemaVersion":"http://datacite.org/schema/kernel-4"}
+       }
+    }
+    '
+  end
+  let(:mocked_headers) do
+    { 'Accept' => '*/*', 'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+      'Content-Type' => 'application/vnd.api+json', 'User-Agent' => 'Faraday v1.1.0' }
+  end
+  let(:mocked_headers_no_content) do
+    { 'Accept' => '*/*', 'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+      'User-Agent' => 'Faraday v1.1.0' }
+  end
+  let(:no_metadata_response_body_json) do
+    '{"data":
+       {"id":"10.33555/tb9q-qb07",
+        "type":"dois",
+        "attributes":
+          {"doi":"10.33555/tb9q-qb07"}
+       }
+    }'
+  end
+  let(:doi_not_found_response_body_json) do
+    "{\"errors\":
+       [{\"status\":\"404\",\"title\":\"The resource you are looking for does'nt exist.\"}]
+    }"
+  end
+
+  before do
+    DATACITE[:rest_api] = 'https://api.test.datacite.org'
+    DATACITE[:user] = 'FriendlyUser'
+    DATACITE[:password] = 'FriendlyPassword'
+    DATACITE[:prefix] = '10.33555'
+  end
```

Comment:
This is what I meant before about testing being easier if you don't use constants in your class.  I don't believe that global constants are reset between tests.

If you have to change their values in a test, the recommended way is via stubbing rather than reassign them: https://relishapp.com/rspec/rspec-mocks/docs/mutating-constants

Reply:
`before` clause in question has been removed.